### PR TITLE
Add progress bar to execute --how tmt

### PR DESCRIPTION
**Pull request by FrNecas**
Resolves: #662

### CODE REVIEW

**Review by jscotka — COMMENTED** [review]
thanks a lot. it seems exactly what I've expected. also this cmd line option is fine.

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for writing this. This will be useful! Tested with `tmt run plan -n core`. When there's a test with a shorter name following one with longer name their names seem to overlap:

```
    execute
        how: tmt
            (3/11) /tests/core/drysst
```

Would it be possible to clear the name of previous test? Also could we erase the last test result so that the overal output is clean?

One more thought: Have you considered using click's [progress bar](https://click.palletsprojects.com/en/5.x/utils/#showing-progress-bars)? Not sure how it would look like though. Just brainstorming.

**Comment by lukaszachy** [comment]
In the case it helps - click doesn't print progress bar immediately as there is check for delay between bar initialization and first attempt to print progress. This looks fixed on github for 8.x but not yet on fedora/pypi where is 7.x
So I hacked around it by setting to `bar.short_limit` to a very low number so https://github.com/pallets/click/blob/7.1.2/src/click/_termui_impl.py#L263 is True.
Otherwise it is quite nice & easy way to print progress bar.
BTW, since we know how long is max. time (sum of all durations) we can print ETA, right? :)

**Comment by FrNecas** [comment]
I've considered creating a status bar that would look similar to the one from click (basically `[####   ]`) but imho in the context of tmt, showing the current test and its index makes a bit more sense. Although that ETA is nice, I think it would be way off in the case of tmt..

Thanks for the bug reports, completely removing the progress should be easy. As for the overlap, I think the only solution is adding enough spaces to the end of the status line that it completely overrides the previous one. Will try to implement this

**Comment by FrNecas** [comment]
@psss Both of the issues should be fixed now

File: tmt/steps/execute/internal.py [hunk 1]
```diff
@@ -52,21 +52,38 @@ def wake(self):
...
                 and not self.opt('debug'):
             # Only show progress bar in an interactive shell.
+            try:
+                # We need to completely override the previous message, add
+                # spaces if necessary.
+                message = message.ljust(len(self.previous_progress_message))
+            except AttributeError:
+                # First iteration, previous message not set
+                pass
+            self.previous_progress_message = message
```
Comment (lukaszachy) [inline tmt/steps/execute/internal.py:80]: Could it be a private attribute? It is not supposed to be used outside of this method (which is already private), right?

Comment (psss) [reply tmt/steps/execute/internal.py]: Makes sense, covered in 7619218.

**Review by psss — APPROVED** [review]
Thanks for fixing the issues. I added a couple of adjustments in 7619218. Please have a look if everything looks good to you. Thanks.

**Comment by psss** [comment]
I propose to use square brackets for the progress to make it distinct from timeouts:

```
    execute
        how: tmt
            00:00:00 /test/good/beakerlib [1/4]
            00:00:01 /test/good/shell [2/4]
            00:00:10 /test/long/beakerlib [3/4] (timeout)
            00:00:03 /test/long/shell [4/4] (timeout)
        summary: 4 tests executed
```

**Comment by FrNecas** [comment]
Using square brackets definitely makes sense and makes the output more clear, +1 from me. LGTM

**Comment by psss** [comment]
Good, thanks!

**Pull request merged**